spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -1152,8 +1152,32 @@ If the participant has already responded successfully to an `@Compensate`
 or `@Complete` method invocation then it MAY report `410 Gone`
 HTTP status code or in the case of non-JAX-RS method returning
 <<source-ParticipantStatus,ParticipantStatus>> `null`.
-This enables the participant to free up resources if it
-hasn't done so previously.
+
+Let's provide some intuition on this matter with the following example.
+
+Let's have an LRA participant containing methods annotated with `@Compensate` and `@Status`.
+The LRA is cancelled and the LRA implementation tries to invoke the participant's `@Compensate` method.
+At that time a network failure happens. The LRA implementation receives a network error as the result of the call.
+Now it cannot be sure if the `@Compensate` call passed through and participant has already compensated the work
+and only the response was lost. Or if the `@Compensate` call has not reached the participant at all.
+As the participant defines the `@Status` method the LRA implementation MUST invoke it to find the participant's true status
+(invocation can be processed several times until succeeds).
+The participant's `@Status` method is expected to indicates whether the `@Compensate` was successfully processed.
+
+* If the participant processed the `@Compensate` call it may <<jaxrs-participant-methods, has already forgotten about the LRA identification>>.
+  Then for JAX-RS participant, the response is `410 Gone`. For non-JAX-RS participant, the method returns `null`.
+  (The implementation MUST assume that all required actions are performed and the compensate invocation MUST NOT be repeated.)
```

Comment:
In some network setups requests are automatically replayed (MicroProfile FT for example) so I would not be this definitive.

Reply:
These invocations are performed by the implementation so I wouldn't expect technologies like FT would be in place. This MUST be definitive because we give a guarantee with the Status method that Compensate doesn't need to be idempotent and thus it shouldn't be replayed.

Reply:
You cannot guarantee that they won't be replayed.

Reply:
I agree with the @mmusgrov  that we can't guarantee the replay does happen. The text should be written from the perspective of the specification. I rephrased it.

As we don't know if the system uses some replay processing framework and as the message order can't be ensured (like compensate http call is invoked, there is some failure in the network and the call is buffered on some router, the initial invocation may timed-out (or coordinator crashes) and the status+compensate invocation is repeated, then suddenly the initial http request could be processed on the router and the duplicity occurs).

I didn't want to make the text complex initially, now I added a note from this perspective at the end.

Reply:
After discussing and re-thinking the point I'm leaving just the initial text without any additional note.

My perspective remains here - with status method the implementation has to guarantee that the invocation is not  repeated. The text is already rewritten in this regard.
I agree with @xstefank that specification needs to be definitive in this regard.

I'm in agreement with @mmusgrov's comment (if I have the correct understanding) that specification can't guarantee the replay does not happen. But the specification must guarantee that it's not replayed by the implementation. As mentioned, the text was rewritten in this regard.